src/util/InitUpdateButton.cpp

```diff
@@ -14,8 +14,17 @@ void initUpdateButton(Button &button,
     QObject::connect(&button, &Button::leftClicked, [&button] {
         auto dialog = new UpdateDialog();
         dialog->setActionOnFocusLoss(BaseWindow::Delete);
-        dialog->move(button.mapToGlobal(
-            QPoint(int(-100 * button.scale()), button.height())));
+
+        auto globalPoint = button.mapToGlobal(
+            QPoint(int(-100 * button.scale()), button.height()));
+
+        // Make sure that update dialog will not go off left edge of screen
+        if (globalPoint.x() < 0)
+        {
+            globalPoint.setX(0);
+        }
+
+        dialog->move(globalPoint);
```

Comment:
```suggestion
        dialog->move(button.mapToGlobal(
            QPoint(std::max(0, int(-100 * button.scale())), button.height())));
```

Reply:
Unfortunately this is hard to make work cleanly while keeping the 100px offset because the x-coordinate must be max'd with 0 _after_ mapping to global coordinates, which (to my knowledge) can only be done through the getter and setter. 
`std::max(0, int(-100 * somePositiveValue))` is always 0 so the popup will always match the edge of the update button:
<img width="400" alt="pic" src="https://user-images.githubusercontent.com/24928223/80154070-d85b5000-858c-11ea-8207-06e1a9850ebf.png">
To be honest I don't really see the importance of the 100px offset but I didn't want to change the design too much.

Reply:
Hmm, you are right, sorry.